plugins/inputs/couchbase/couchbase.go

```diff
@@ -379,13 +391,39 @@ func (cb *Couchbase) Init() error {
 
 	cb.bucketInclude = f
 
+	tlsConfig, err := cb.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	cb.client = &http.Client{
+		Timeout: 10 * time.Second,
+		Transport: &http.Transport{
+			MaxIdleConnsPerHost: couchbaseClient.MaxIdleConnsPerHost,
+			TLSClientConfig:     tlsConfig,
+		},
+	}
+
+	// Couchbase requires that a TLS cert and key be provided when insecure skip verify is disabled
+	if !cb.ClientConfig.InsecureSkipVerify && (cb.ClientConfig.TLSCert == "" || cb.ClientConfig.TLSKey == "") {
+		return fmt.Errorf("tls_cert and tls_key are required when insecure_skip_verify = false")
+	}
+
+	couchbaseClient.SetSkipVerify(cb.ClientConfig.InsecureSkipVerify)
+	couchbaseClient.SetCertFile(cb.ClientConfig.TLSCert)
+	couchbaseClient.SetKeyFile(cb.ClientConfig.TLSKey)
+	couchbaseClient.SetRootFile(cb.ClientConfig.TLSCA)
+
 	return nil
 }
 
 func init() {
 	inputs.Add("couchbase", func() telegraf.Input {
 		return &Couchbase{
 			BucketStatsIncluded: []string{"quota_percent_used", "ops_per_sec", "disk_fetches", "item_count", "disk_used", "data_used", "mem_used"},
+			ClientConfig: tls.ClientConfig{
+				InsecureSkipVerify: true,
```

Comment:
We can't skip tls verification by default. I'd prefer to default to false and then provide a helpful error message at runtime that gives the user the two options: 1. provide cert and key or 2. skip tls verification by setting insecure_skip_verify true. That way the user has to choose the insecure option and not us.

Reply:
I agree it would be best to default to false. However, couchbase defaults to skipping TLS verification, so I was trying to keep it backward compatible.

Reply:
So we use go-couchbase for everything but queryDetailedBucketStats, and go-couchbase skips verification by default? I looked through go-couchbase and see the same thing. They don't even provide a way to check what their default is. I would prefer to get our default from them so if they change in the future we won't be left defaulting to skipping, but that's not possible.

Reply:
If we **have to** reproduce their default it's also ok for me to have an `insecure_skip_verify` option defaulting to `true` with a big fat warning! I just want to make sure the user _can_ choose to be secure. :-)

Reply:
skipping verification is a bad default, and if go-couchbase is doing it, we don't want to copy that. This should be:
```suggestion
				InsecureSkipVerify: false,
```
here and in the documentation.

Reply:
This would also be my preferred setting. :-)